From 69b268dc2337fff25844a246de7dcbe10fa658a0 Mon Sep 17 00:00:00 2001
From: Peter Ogden <ogden@xilinx.com>
Date: Tue, 24 Sep 2019 07:11:30 -0400
Subject: [PATCH] Revert "net: phy: add GBit master / slave error detection"

The PHY on the Pynq-Z2 board signals a Autonegotiaion failure
if there is no cable attached resulting in the PHY being disabled
in normal operation.

This reverts commit b8f8c8eb408b36ad55dd41a616b3f51998880fb6.
---
 drivers/net/phy/phy_device.c | 8 --------
 include/uapi/linux/mii.h     | 1 -
 2 files changed, 9 deletions(-)

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index 19ab8a7..0531df0 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -1561,14 +1561,6 @@ int genphy_read_status(struct phy_device *phydev)
 			if (adv < 0)
 				return adv;
 
-			if (lpagb & LPA_1000MSFAIL) {
-				if (adv & CTL1000_ENABLE_MASTER)
-					phydev_err(phydev, "Master/Slave resolution failed, maybe conflicting manual settings?\n");
-				else
-					phydev_err(phydev, "Master/Slave resolution failed\n");
-				return -ENOLINK;
-			}
-
 			phydev->lp_advertising =
 				mii_stat1000_to_ethtool_lpa_t(lpagb);
 			common_adv_gb = lpagb & adv << 2;
diff --git a/include/uapi/linux/mii.h b/include/uapi/linux/mii.h
index a506216..b5c2fdc 100644
--- a/include/uapi/linux/mii.h
+++ b/include/uapi/linux/mii.h
@@ -136,7 +136,6 @@
 #define CTL1000_ENABLE_MASTER	0x1000
 
 /* 1000BASE-T Status register */
-#define LPA_1000MSFAIL		0x8000	/* Master/Slave resolution failure */
 #define LPA_1000LOCALRXOK	0x2000	/* Link partner local receiver status */
 #define LPA_1000REMRXOK		0x1000	/* Link partner remote receiver status */
 #define LPA_1000FULL		0x0800	/* Link partner 1000BASE-T full duplex */
-- 
2.7.4

